src/encoded/commands/create_mapping_on_deploy.py

```diff
@@ -144,8 +145,13 @@ def main():
         env = app.registry.settings.get('env.name')
         if 'webprod' in env:
             if data_env != env:
-                log.info("looks like we are on staging, run create mapping without check first")
-                run_create_mapping(app, check_first=False, item_order=ITEM_INDEX_ORDER)
+                log.info("Looks like we are deploying staging, checking --wipe-es")
```

Comment:
Hmm. In my view, this logic about how we tell what we're deploying to needs to be cleaned up to be less ad hoc in nature, but that wouldn't be a necessarily local change. The minimal thing would be to isolate the `data_env != env` logic in a separate function that knows what the accepted technique is, just in case other things need to ask this question so they all use the same mechanism. A better strategy would be to declare the intent that this be staging and have the choice to separate that data env drive off that declaration and poll that same piece of declarative information here. But for now the technique here looks good enough and uses a reasonable heuristic.

I might prefer the message say something like `"Looks like we are not using the production data store,..."` rather than `"Looks like we are deploying staging,..."` so it isn't saying something that is stronger than what it knows. It also might be ok to say "Looks like this is a staging deploy,..." (note change to "a staging deploy" rather than "deploying staging", which would suggest there were different deploys that count, rather than it has verified a domain name or something. But this is a smallish concern and need not hold anything up now. I just find that sometimes if something goes wrong, people read these messages and think that something has been checked that hasn't been, and it can throw them off the track of what they should be diving further into.